Sources/MongoSwift/MongoCollection+FindAndModify.swift

```diff
@@ -0,0 +1,127 @@
+
+/// An extension of `MongoCollection` encapsulating find and modify operations.
+extension MongoCollection {
+    /**
+     * Finds a single document and deletes it, returning the original.
+     *
+     * - Parameters:
+     *   - filter: `Document` representing the match criteria
+     *   - options: Optional `FindOneAndDeleteOptions` to use when executing the command
+     *
+     * - Returns: The deleted document, or `nil` if no document was deleted.
+     */
+    @discardableResult
+    public func findOneAndDelete(_ filter: Document, options: FindOneAndDeleteOptions? = nil) throws -> Document? {
+        throw MongoError.commandError(message: "Unimplemented command")
+    }
+
+    /**
+     * Finds a single document and replaces it, returning either the original or the replaced document.
+     * 
+     * - Parameters:
+     *   - filter: `Document` representing the match criteria
+     *   - replacement: a `Document` to replace the found one
+     *   - options: Optional `FindOneAndReplaceOptions` to use when executing the command
+     *
+     * - Returns: Either the original document or its replacement depending on selected options, 
+     *            or `nil` if there was no match.
+     */
+    @discardableResult
+    public func findOneAndReplace(filter: Document, replacement: Document,
+                                  options: FindOneAndDeleteOptions? = nil) throws -> Document? {
+        throw MongoError.commandError(message: "Unimplemented command")
+    }
+
+    /**
+     * Finds a single document and updates it, returning either the original or the updated document.
+     * 
+     * - Parameters:
+     *   - filter: `Document` representing the match criteria
+     *   - replacement: a `Document` to replace the found one
+     *   - options: Optional `FindOneAndUpdateOptions` to use when executing the command
+     *
+     * - Returns: Either the original or updated document depending on selected options, 
+     *            or `nil` if there was no match.
+     */
+    @discardableResult
+    public func findOneAndUpdate(filter: Document, update: Document,
+                                 options: FindOneAndUpdateOptions? = nil) throws -> Document? {
+        throw MongoError.commandError(message: "Unimplemented command")
+    }
+}
+
+/// Indicates which document to return in a find and modify operation.
+public enum ReturnDocument {
+    /// Indicates to return the document before the update, replacement, or insert occured.
+    case before
+
+    ///  Indicates to return the document after the update, replacement, or insert occured.
+    case after
+}
+
+/// Options to use when executing a `findOneAndDelete` command on a `MongoCollection`. 
+public struct FindOneAndDeleteOptions {
+    /// Specifies a collation to use.
+    public let collation: Document?
+
+    /// The maximum amount of time to allow the query to run.
+    public let maxTimeMS: Int64?
+
+    /// Limits the fields to return for all matching documents.
```

Comment:
Small nit, but "documents" can refer to "document" for all descriptions in the `findAndModify` option structs since these operations never work with multiple documents.